### PR TITLE
Config/gregtech.cfg

### PR DESCRIPTION
**Pull request by boubou19**
- migrate the content of `GregTech/gregtech.cfg` to the GTNHLib config system.
- migrate cleanroom related blocks to its own `Cleanroom.cfg`
- migrate underground fluids to its own `UndergroundFluids.cfg`
- created a new builder to add/modify easily underground fluids for the default config
- tuned the default config to look like the one in the modpack repo
- `crashOnNullRecipeInput` now does nothing in dev env, you have to use the debug flags instead
- Fixed `mPollutionOnExplosion` not being generated because of a typo made 3 years ago (by me lol) which caused the key to not generate in the config and it was using the sour rain threshold as a value, which caused the machines to generate 2M gibbl per machine explosion instead of 100k.
- Fixed config not generating config for chain wrench on pipes and chain spray on blocks. 
- Moved the galactic greg ore dim registration from NHCore to GT5U (corresponding PR: https://github.com/GTNewHorizons/NewHorizonsCoreMod/pull/944)
- Used Enums to ease the registration of the mod containers, the dimensions, and the blocks in the world gen.
- Used enums to ease the definition of ore mixes, small ores, and stone veins.
- **Removed the bartwork compat to galactic greg, as the only use for it was 100% config driven ore mixes / small ores, and the config never used such thing.**
- Replaced `Unifications.cfg` with `gregtech.api.enum.ManualOreDictTweaks`. I didn't found a good place to put it, so i placed  it there despite not being an enum.
- Fixed planet dusts being null in EoH recipes in dev. Now they fallback to stone dust if NHCore is not loaded.
- Ported GT NEI Ore Plugin config to GTNH Lib config system.
- Moved generated CSVs from `<instance>/config` to `<instance>`
- Fixed the CSV generator to avoid some column shifts.

removed configs:
- `GregTech.cfg`
  - `BrickedBlastFurnace` (wasn't used at all)
  - `"GTNH Optimised Material Loading"` (it's an actual optimisation for a mod used only in the pack, so now it's hardcoded)
  - `disable_STDOUT` (was disabling ALL the logs, no one sane will ever do this)
  - `constant_need_of_energy` (wasn't used at all)
  
- `MachineStats.cfg`
    - Purged a ton of configs that shouldn't be (for instance, the efficiency of a gas generator is defined in code, but was configurable. Now it's not the case anymore)

- `MaterialProperties.cfg`
  - Yeeted the whole file as we can directly tweak the Material enum if needed, so it's pure bloat (moved the harvest level config to `GregTech.cfg` under the `harvest_level` category)
- `OverpoweredStuff.cfg`
  - removed `WireHeatingTicks` as it seems it's a leftover of some removed feature concerning cable heat (also removed its usage as it was displayed when you use the portable scanner on a cable)
  - removed `EnableZPMandUVBatteries` as it seems they are enabled in full pack.
  - removed the whole `oreprocessingoutputmultiplier` section, it's pure bloat as we don't use it in full pack.
- `Other.cfg`
  - `AprilFool` config. It was disabled and not used anymore ("was turning all the textures into trololo ones) 
  - `EnableLagencyOilGalactiCraft` config. It was disabled since a very long time.
  - `DisableFlintTools` config. flint tools are required in pack progression.
- `Client.cfg`
  - `CircuitsOrder` Seems to solve an issue in an incredibly hacky way, and is not used in the modpack repo since its introduction in 2022.
- `WorldGeneration.cfg`
  - yeeted the `worldgen.ore.mix` category, and it's now replaced by a nice enum if ore mixes need edits: `gregtech.api.enums.OreMixes`.
  - yeeted the `worldgen.ore.small` category, and it's now replaced by a nice enum if small ores need edits: `gregtech.api.enums.SmallOres`.
  -  yeeted `nether.stone` and `overworld.stone` categories, and it's now replaced by a nice enum if GT stone gens need edits: `gregtech.api.enums.GTStones`.
- `Unifications.cfg`
  - Yeeted the whole file. Now edit `gregtech.api.enum.ManualOreDictTweaks` if you need to mess with oredict in other mods. 

I need someone to check if i didn't derped in this [commit](https://github.com/GTNewHorizons/GT5-Unofficial/pull/2939/commits/d2a0c5e7f6edc07ed47157e404d13c93e77b65d8), i'm not familiar with the Material system enough.

### CODE REVIEW

File: src/main/java/gregtech/api/world/GT_Worldgen.java [hunk 1]
```diff
@@ -112,20 +86,32 @@ public boolean isGenerationAllowed(World aWorld, int aAllowedDimensionType) {
      */
     public boolean isGenerationAllowed(World aWorld, Class... aAllowedDimensionTypes) {
         String aDimName = aWorld.provider.getDimensionName();
+        if (aDimName.equalsIgnoreCase("Underdark")) {
+            return false;
+        }
```
Comment (Caedis) [inline src/main/java/gregtech/api/world/GT_Worldgen.java:91]: Is this intended? Old logic is to not gen in the Underdark

File: src/main/java/gregtech/common/config/gregtech/ConfigOreDropBehavior.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @Config.Comment({ "Settings:",
+        " - 'PerDimBlock': Sets the drop to the block variant of the ore block based on dimension, defaults to stone type",
+        " - 'UnifiedBlock': Sets the drop to the stone variant of the ore block",
+        " - 'Block': Sets the drop to the ore  mined",
+        " - 'FortuneItem': Sets the drop to the new ore item and makes it affected by fortune"
+            + " - 'Item': Sets the drop to the new ore item" })
+
+    @Config.DefaultString("FortuneItem")
+    @Config.RequiresMcRestart
+    public static String setting;
```
Comment (Caedis) [inline src/main/java/gregtech/common/config/gregtech/ConfigOreDropBehavior.java:23]: Does DefaultEnum not work?

File: src/main/java/gregtech/api/world/GT_Worldgen.java (hunk 1 above)
Comment (boubou19) [reply src/main/java/gregtech/api/world/GT_Worldgen.java:91]: Old logic says that if it's not a dim from DD, TF, OW, End or Nether, it is not allowed.Then it performs additional checks  to determine if it's allowed.We want DD to be excluded no matter what, so if DD is found, return false.

File: src/main/java/gregtech/common/config/gregtech/ConfigOreDropBehavior.java (hunk 2 above)
Comment (boubou19) [reply src/main/java/gregtech/common/config/gregtech/ConfigOreDropBehavior.java:23]: I did not understand how to use it tbh.But it's equivalent.

File: src/main/java/gregtech/api/world/GT_Worldgen.java (hunk 1 above)
Comment (Caedis) [reply src/main/java/gregtech/api/world/GT_Worldgen.java:91]: Gotcha

**Review by Caedis — APPROVED** [review]

**Pull request merged**